contracts/src/CrossChainApplications/examples/NativeTokenBridge/TokenSource.sol

```diff
@@ -74,31 +71,13 @@ abstract contract TokenSource is ITokenSource, TeleporterOwnerUpgradeable {
             originSenderAddress == nativeTokenDestinationAddress, "TokenSource: unauthorized sender"
         );
 
-        // Decode the payload to recover the action and corresponding function parameters
-        (SourceAction action, bytes memory actionData) = abi.decode(message, (SourceAction, bytes));
-
         // Route to the appropriate function.
-        if (action == SourceAction.Unlock) {
-            (address recipient, uint256 amount) = abi.decode(actionData, (address, uint256));
-            _unlockTokens(recipient, amount);
-        } else if (action == SourceAction.Burn) {
-            uint256 newBurnTotal = abi.decode(actionData, (uint256));
-            _handleBurnTokens(newBurnTotal);
-        } else {
-            revert("TokenSource: invalid action");
-        }
+        (address recipient, uint256 amount) = abi.decode(message, (address, uint256));
+        _unlockTokens(recipient, amount);
```

Comment:
We may want to rename this function to something more general since it now handles unlocks and burns (even though burning is implicit by sending to the burn address). Perhaps `_transferTokens`

Reply:
I think `unlock` is the most general. We've locked the tokens, and are being told to unlock them to an address by the destination. We don't really care if that chain is "burning" them or not